numba/transforms.py

```diff
@@ -86,6 +86,11 @@ def _loop_lift_get_candidate_infos(cfg, blocks, livemap):
     loops = _extract_loop_lifting_candidates(cfg, blocks)
     loopinfos = []
     for loop in loops:
+        # this is to prevent a bad rewrite where a prelude for a lifted
+        # loop would get written into block -1 if a loop entry were in block 0
+        if 0 in loop.entries:
+            continue
```

Comment:
Not 100% convinced by this yet!

Reply:
You can use `cfg.entry_point` to find the entry block.  Also this checks should be moved to https://github.com/numba/numba/blob/ca731bf94bc6b23cdc865470b9e98ef647c0b1c8/numba/transforms.py#L14-L47

Reply:
Good point. I've moved it.